src/pages/Main.tsx

```diff
@@ -1,7 +1,70 @@
-import { Button } from "@deskpro/app-sdk";
+/* eslint-disable react-hooks/exhaustive-deps */
+import { TextArea, GetStateResponse } from "@deskpro/app-sdk";
+import {
+  Stack,
+  Button,
+  H1,
+  useInitialisedDeskproAppClient,
+  useDeskproAppClient,
+} from "@deskpro/app-sdk";
+import { useState } from "react";
+
+import useDebounce from "../utils/debounce";
 
 export const Main = () => {
-    return (
-        <Button text="Greet" onClick={() => alert("Hello")} />
-    );
+  const { client } = useDeskproAppClient();
+
+  const [text, setText] = useState<string>("");
+  const debouncedText = useDebounce(text, 200);
+
+  useInitialisedDeskproAppClient((client) => {
+    client
+      ?.getUserState<string>("scratchpad")
+      .then((res: GetStateResponse<string>[]) => {
+        setText(res[0]?.data ?? "");
+      });
+  });
+
+  useInitialisedDeskproAppClient(
+    (client) => {
+      debouncedText &&
```

Comment:
Will this still work if the user deletes all of the text from the textarea? I.e. they select all of the text and press backspace.

We want to allow the user to manually delete all content from the textarea and the app save an empty string in this case.

Reply:
Please check the latest commit